src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -398,6 +398,25 @@ public void deleteControlledResource(
     }
   }
 
+  /**
+   * List all controlled resources ID in Sam this user has access to. This builds an aggregated list
+   * across the four types of controlled resources.
+   */
+  @Traced
+  public List<UUID> listControlledResourceIds(
+      AuthenticatedUserRequest userReq, String controlledResourceName) {
+    ResourcesApi resourceApi = samResourcesApi(userReq.getRequiredToken());
+    List<UUID> workspaceIds = new ArrayList<>();
+    try {
+      for (var resourceAndPolicy : resourceApi.listResourcesAndPolicies(controlledResourceName)) {
```

Comment:
this needs to call `listResourcesAndPoliciesV2`. Otherwise, the response won't include the resources a user has inherited permission on, which is most controlled resource